core/workspace.js

```diff
@@ -276,11 +272,15 @@ Blockly.Workspace.prototype.renameVariableInternal_ = function(
  * variable to rename with the given name.
  * @param {string} oldName Variable to rename.
  * @param {string} newName New variable name.
+ * @param {string=} opt_type The type of the variable.  If not provided it
+ *     defaults to the empty string, which is a specific type.
  */
-Blockly.Workspace.prototype.renameVariable = function(oldName, newName) {
```

Comment:
TODO Deprecate in favor of passing in the variable + new name

Reply:
Yes.  I may be able to fully delete these functions if I create the variables early enough, and only use ID.

This is already a wrapper on `renameVariableInternal`, which takes in the variable + new name.

Reply:
Added some TODOs.